### DISCUSSION TITLE
Enhancement request: API and UI support for extracting datasource configuration as YAML

### DISCUSSION THREAD

**Discussion by eamonryan** [grafana/grafana#38250] [Dashboards, Panels & Visualizations]
**What would you like to be added**: I would like to see an API endpoint and UI button on datasource configuration pages for extracting/downloading the configuration of a given datasource as YAML for use with the Grafana provisioning feature.

**Why is this needed**: Datasource provisioning requires that the configuration be in YAML, but the API only returns the datasource config in JSON so a user has to manually convert them (and sometimes edit the fields/structure some) which is an unnecessary manual extra step.

Making it so that the downloaded YAML can be immediately used as a provisioning file would add a great deal of convenience.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
